doc/releases/migration-guide-3.6.rst

```diff
@@ -360,6 +360,9 @@ Bluetooth
 * The :c:func:`bt_l2cap_chan_send` API now requires the application to reserve
   enough bytes for the L2CAP headers. Call ``net_buf_reserve(buf,
   BT_L2CAP_SDU_CHAN_SEND_RESERVE);`` at buffer allocation time to do so.
+* Added ``recycled()`` callback to :c:struct:`bt_conn_cb`, which notifies listeners when a
+  connection object has been freed, so it can be utilized for different purposes. No guarantees are
+  made to what listener will be granted the object, as only the first claim is served.
```

Comment:
This looks like a release note. There is a separate document for this. It's a sibling to this document in the tree.

Reply:
I see - I'll move this entry to `doc/releases/release-notes-3.6.rst` if that's okay!

Reply:
To elaborate why: I think migration guide is for "We have some breaking changes. You have to ensure this thing when upgrading, and here's some simplest way to do it that will work for 99% of use-sites." New features in that context are noise that would degrade the quality of the migration guide.